lighthouse-core/config/index.js

```diff
@@ -0,0 +1,162 @@
+
+function filterPasses(passes, audits) {
+  const requiredGatherers = getGatherersNeededByAudits(audits);
+
+  // Make sure we only have the gatherers that are needed by the audits
+  // that have been listed in the config.
+  const filteredPasses = passes.map(pass => {
+    pass.gatherers = pass.gatherers.filter(gatherer => {
+      try {
+        const GathererClass = Driver.getGathererClass(gatherer);
+        return requiredGatherers.has(GathererClass.name);
+      } catch (requireError) {
+        throw new Error(`Unable to locate gatherer: ${gatherer}`);
+      }
+    });
+
+    return pass;
+  })
+
+  // Now remove any passes which no longer have gatherers.
+  .filter(p => p.gatherers.length > 0);
+  return filteredPasses;
+}
+
+function getGatherersNeededByAudits(audits) {
+  // It's possible we didn't get given any audits (but existing audit results), in which case
+  // there is no need to do any work here.
+  if (!audits) {
+    return new Set();
+  }
+
+  return audits.reduce((list, audit) => {
+    audit.meta.requiredArtifacts.forEach(artifact => list.add(artifact));
+    return list;
+  }, new Set());
+}
+
+function filterAudits(audits, whitelist) {
+  // If there is no whitelist, assume all.
+  if (!whitelist) {
+    return Array.from(audits);
+  }
+
+  const rejected = [];
+  const filteredAudits = audits.filter(a => {
+    const auditName = a.toLowerCase();
+    const inWhitelist = whitelist.has(auditName);
+
+    if (!inWhitelist) {
+      rejected.push(auditName);
+    }
+
+    return inWhitelist;
+  });
+
+  if (rejected.length) {
+    log.log('info', 'Running these audits:', `${filteredAudits.join(', ')}`);
+    log.log('info', 'Skipping these audits:', `${rejected.join(', ')}`);
+  }
+
+  return filteredAudits;
+}
+
+function expandAudits(audits) {
+  return audits.map(audit => {
+    try {
+      return require(`../audits/${audit}`);
+    } catch (requireError) {
+      throw new Error(`Unable to locate audit: ${audit}`);
+    }
+  });
+}
+
+function expandArtifacts(artifacts) {
+  const expandedArtifacts = Object.assign({}, artifacts);
+
+  // currently only trace logs and performance logs should be imported
+  if (expandedArtifacts.traceContents) {
+    expandedArtifacts.traceContents = require(artifacts.traceContents);
+  }
+  if (expandedArtifacts.performanceLog) {
+    expandedArtifacts.CriticalRequestChains =
+      parsePerformanceLog(require(artifacts.performanceLog));
+  }
+
+  return expandedArtifacts;
+}
+
+function parsePerformanceLog(logs) {
+  // Parse logs for network events
+  const networkRecords = recordsFromLogs(logs);
+
+  // Use critical request chains gatherer to create the critical request chains artifact
+  const criticalRequestChainsGatherer = new CriticalRequestChainsGatherer();
+  criticalRequestChainsGatherer.afterPass({}, {networkRecords});
+
+  return criticalRequestChainsGatherer.artifact;
+}
+
+/**
+ * @return {!Config}
+ */
+class Config {
+  constructor(config, whitelist) {
```

Comment:
maybe call the parameters `configJSON` and `auditWhitelist` to make the distinctions clear? Also should add a doc comment for the constructor